tests/integration/ha/test_replication.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+import time
+
+import pytest
+from ha_helpers import (
+    APP_NAME,
+    REL_NAME_ADMIN,
+    ZK_NAME,
+    check_logs,
+    get_topic_leader,
+    kill_unit_process,
+    produce_and_check_logs,
+)
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+DUMMY_NAME = "app"
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest, kafka_charm, app_charm):
+    await asyncio.gather(
+        ops_test.model.deploy(kafka_charm, application_name=APP_NAME, num_units=1, series="jammy"),
+        ops_test.model.deploy(ZK_NAME, channel="edge", num_units=1, series="jammy"),
+        ops_test.model.deploy(app_charm, application_name=DUMMY_NAME, num_units=1, series="jammy"),
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, ZK_NAME, DUMMY_NAME],
+        idle_period=30,
+        timeout=3600,
+    )
+    assert ops_test.model.applications[APP_NAME].status == "blocked"
+    assert ops_test.model.applications[ZK_NAME].status == "active"
+
+    await ops_test.model.add_relation(APP_NAME, ZK_NAME)
+    async with ops_test.fast_forward():
+        await ops_test.model.wait_for_idle(apps=[APP_NAME, ZK_NAME], idle_period=30)
+        assert ops_test.model.applications[APP_NAME].status == "active"
+        assert ops_test.model.applications[ZK_NAME].status == "active"
+
+    await ops_test.model.add_relation(APP_NAME, f"{DUMMY_NAME}:{REL_NAME_ADMIN}")
+    await ops_test.model.wait_for_idle(apps=[APP_NAME, DUMMY_NAME], idle_period=30)
+    assert ops_test.model.applications[APP_NAME].status == "active"
+    assert ops_test.model.applications[DUMMY_NAME].status == "active"
+
+
+async def test_second_cluster(ops_test: OpsTest, kafka_charm):
+    second_kafka_name = f"{APP_NAME}-two"
+    second_zk_name = f"{ZK_NAME}-two"
```

Comment:
Hm, yes in general. But not in this case, because I'm explicitly not marking this with `skip_if_deployed`, the second cluster needs to be deployed. But I understand the reasoning behind this. In any case, it's out of the scope of this PR, as it affects literally all tests we have at the moment.